DefaultUrlMappingInfo.toString is NPE unsafe.

I am seeing an NPE in org.grails.web.errors.GrailsExceptionResolver#forwardRequest when it is trying to debug log contents of UrlMappingInfo info. 

While debugging and trying to understand GrailsExceptionResolver I am seeing following exception:
```
Caused by: java.lang.NullPointerException: null
        at org.grails.web.mapping.DefaultUrlMappingInfo.toString(DefaultUrlMappingInfo.java:161)
        at java.lang.String.valueOf(String.java:2994)
        at java.lang.StringBuilder.append(StringBuilder.java:131)
        at org.grails.web.errors.GrailsExceptionResolver.forwardRequest(GrailsExceptionResolver.java:218)
        at org.grails.web.errors.GrailsExceptionResolver.resolveViewOrForward(GrailsExceptionResolver.java:199)
```
See https://github.com/grails/grails-core/blob/7d464dad2a314c02bb175d6e8772a2aca302ca7c/grails-web-mvc/src/main/groovy/org/grails/web/errors/GrailsExceptionResolver.java#L218

This occurs because info.toString() can throw NPE if the object is DefaultUrlMappingInfo which has null urlData field.

https://github.com/grails/grails-core/blob/7ab9e47ad805fbeb9433a488dd33f91bef44c0fa/grails-web-url-mappings/src/main/groovy/org/grails/web/mapping/DefaultUrlMappingInfo.java#L161

As a mitigation I had to turn off the DEBUG logging for GrailsExceptionResolver class.

--- a/grails-web-url-mappings/src/main/groovy/org/grails/web/mapping/DefaultUrlMappingInfo.java
+++ b/grails-web-url-mappings/src/main/groovy/org/grails/web/mapping/DefaultUrlMappingInfo.java
@@ -158,6 +158,9 @@ public class DefaultUrlMappingInfo extends AbstractUrlMappingInfo {
 
     @Override
     public String toString() {
+        if (urlData == null) {
+            return null;
+        }
         return urlData.getUrlPattern();
     }
 

